scripts/mods/mod_settings_manager.nut

```diff
@@ -0,0 +1,99 @@
+this.mod_settings_manager <- {
+	m = {
+		Pages = null,
+		Locked = false
+	},
+
+	function create()
+	{
+		this.m.Pages = this.MSU.OrderedMap();
+	}
+	
+	function add( _page )
+	{
+		if (this.m.Locked)
+		{
+			this.logError("Settings Manager is Locked, no more settings can be added");
+		}
+		else
+		{
+			if (!(_page instanceof this.MSU.SettingsPage))
+			{
+				throw this.Exception.InvalidTypeException;
+			}
+			this.m.Pages[_page.getModID()] <- _page;
```

Comment:
As discussed on server use  `this.m.Mods[_page.getModId()][_page.getID()] <- page;`